packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -366,6 +370,15 @@ export default class CosmeticFilterBucket {
     // Tokens from `hostname` and `domain` which will be used to lookup filters
     // from the reverse index. The same tokens are re-used for multiple indices.
     const hostnameTokens = createLookupTokens(hostname, domain);
+
+    let parentHostnameTokens: ReturnType<typeof createLookupTokens>[] | undefined = undefined;
+    if (parentDomains.length !== 0 && parentDomains.length === parentHostnames.length) {
```

Comment:
Understanding question: is the API intended that parentDomains and parentHostnames should generally be of the same length (or otherwise being ignored)?

In that case, would it be less error prone for the caller to pass them in as some form of list of pairs? In other words, some construct that enforces that the length matches?